### DISCUSSION TITLE
Can I make folds start on the *next* line?

### DISCUSSION THREAD

**Discussion by Hubro** [nvim-treesitter/nvim-treesitter#2970] [Q&A]
When folding Rust code, I only want to fold the contents of a block. For example:

```rust
struct LogLine {
    severity: String,
    date: String,
    logger_name: String,
    thread_name: String,
}
```

```
struct_item [3, 0] - [8, 1]
  name: type_identifier [3, 7] - [3, 14]
  body: field_declaration_list [3, 15] - [8, 1]
    field_declaration [4, 4] - [4, 20]
      name: field_identifier [4, 4] - [4, 12]
      type: type_identifier [4, 14] - [4, 20]
    field_declaration [5, 4] - [5, 16]
      name: field_identifier [5, 4] - [5, 8]
      type: type_identifier [5, 10] - [5, 16]
    field_declaration [6, 4] - [6, 23]
      name: field_identifier [6, 4] - [6, 15]
      type: type_identifier [6, 17] - [6, 23]
    field_declaration [7, 4] - [7, 23]
      name: field_identifier [7, 4] - [7, 15]
      type: type_identifier [7, 17] - [7, 23]
```

I want to fold the "field_declaration_list", but unfortunately it starts on the first line, at the opening bracket (`{`) and ends at the line with the closing bracket. This means that the entire struct gets folded into a single line.

Is there any way to make the fold start on the next line, and also end one line earlier?

The effect I'm aiming for:

```rust
struct LogLine {
    severity: String,       // <- These lines get folded into a single line
    date: String,           // <-
    logger_name: String,    // <-
    thread_name: String,    // <-
}
```

**Comment by theHamsta** [comment]
https://github.com/nvim-treesitter/nvim-treesitter/pull/2943 apart from the one line earlier. But we could make the one line earlier an option

**Reply by Hubro** [reply]
Aha, that looks great! Also the timing is quite a coincidence, I just recently started learning Rust and your pull request is barely 2 weeks old :smile:

I added my suggestion as a comment to your PR.
